### PR TITLE
Remove VirtualService examples that no longer have an effect

### PR DESCRIPTION
**Pull request by esnible**
For https://github.com/istio/istio/issues/11891

No longer needed because of https://github.com/istio/istio/pull/11735

### CODE REVIEW

File: samples/external/github.yaml [hunk 1]
```diff
@@ -17,24 +17,6 @@ spec:
     name: https
     protocol: HTTPS
 ---
```
Comment (linsun) [inline samples/external/github.yaml]: remove this too?

File: samples/external/pypi.yaml [hunk 2]
```diff
@@ -17,24 +17,6 @@ spec:
     name: https
     protocol: HTTPS
 ---
```
Comment (linsun) [inline samples/external/pypi.yaml]: remove this extra line?

**Comment by esnible** [comment]
@vadimeisenbergibm Question.  I tested this interactively using "reviews" sample pod.

I understand why I needed a SE for apt-get to install git
I don't understand why using 1.1 snapshot 6 I did *NOT* need a SE for `git clone https://github.com/istio/api.git` but I did need an SE for `pip search istio`.  Both are doing https.  The only difference I could think of was the redirect.

**Comment by vadimeisenbergibm** [comment]
@esnible Previously, if you defined a Service Entry without a VirtualService for TLS traffic, you would get various undefined behavior - accessing any site on this port, or arriving to the same URL as the Service Entry for any site, etc. Starting from https://gcsweb.istio.io/gcs/istio-prerelease/daily-build/release-1.1-20190214-09-16/, you just define a ServiceEntry without a VirtualService for any protocol, and it should work. Otherwise it is a bug.

**Comment by esnible** [comment]
@vadimeisenbergibm I like the behavior of ServiceEntry working without VirtualService.

I am concerned that `git clone` is working without a ServiceEntry for github.com, but that `pip search` is not working without a ServiceEntry for pypi.org.  (The only thing I can think of that is different is that `pip` may need to process a redirect.)  Should I be able to do `pip search istio` without a ServiceEntry?

**Comment by vadimeisenbergibm** [comment]
@esnible 
> `git clone` is working without a ServiceEntry for github.com

it is a bug, please open an issue and specify the release version.

**Comment by vadimeisenbergibm** [comment]
@esnible ^ please check if you experience this behavior with https://gcsweb.istio.io/gcs/istio-prerelease/daily-build/release-1.1-20190214-09-16/

File: samples/external/github.yaml (hunk 1 above)
Comment (esnible) [reply samples/external/github.yaml]: Testing with release-1.1-20190214-09-16 showed this was needed.

**Comment by esnible** [comment]
@vadimeisenbergibm Works as expected with release-1.1-20190214-09-16.

**Comment by vadimeisenbergibm** [comment]
@esnible OK, great!

**Comment by esnible** [comment]
@linsun Can we get this in?  We don't want overly-complex samples.

**Comment by linsun** [comment]
cc @wenchenglu

**Pull request merged**